kernel/devices/LApic.cpp

```diff
@@ -82,13 +101,41 @@ namespace Kernel::Devices
         //TODO: x2APIC support
         
         baseAddress = (CPU::ReadMsr(MSR_APIC_BASE) & ~(0xFFF)) + vmaHighAddr;
+        bool isX2ApicFeaturePresent = CPU::FeatureSupported(CpuFeature::X2APIC);
+        Logf("X2Apic feature status: %b", LogSeverity::Info, isX2ApicFeaturePresent);
         Memory::PageTableManager::Local()->MapMemory(baseAddress, baseAddress - vmaHighAddr, Memory::MemoryMapFlag::AllowWrites);
+        
         apicId = ReadReg(LocalApicRegister::Id) >> 24;
```

Comment:
perhaps we should do this inside the x2 block, since this will be invalid, and if x2 is not available, we do this.
Can you also update GetId() to support getting the x2 apic id too!

Reply:
I updated the GetId function, but left the init use the ReadReg method, is there a specific reason why you did it? Can i switch to use GetId instead?